src/world/mod.rs

```diff
@@ -542,15 +540,23 @@ impl World {
 
         self.resources
             .remove(&id)
-            .map(TrustCell::into_inner)
+            .map(AtomicRefCell::into_inner)
             .map(|x: Box<dyn Resource>| x.downcast())
             .map(|x: Result<Box<R>, _>| x.ok().unwrap())
             .map(|x| *x)
     }
 
     /// Internal function for fetching resources, should only be used if you
     /// know what you're doing.
-    pub fn try_fetch_internal(&self, id: ResourceId) -> Option<&TrustCell<Box<dyn Resource>>> {
+    ///
+    /// # Safety
+    ///
+    /// If this is used to replace the `Box<dyn Resource>` with a different one, the new one must
+    /// have a `TypeId` that matches the one in the `ResourceId` provided here.
+    pub unsafe fn try_fetch_internal(
```

Comment:
Oh almost forgot!

The one other thing I did in these changes is to mark this method as `unsafe` since it would otherwise allow safely swapping out the `Box<dyn Resource>` stored at a particular `ResourceId` with any other arbitrary `Box<dyn Resource>` that doesn't necessarily correspond to the `ResourceId`.